pyk/src/pyk/tests/test_kcfg.py

```diff
@@ -263,11 +258,75 @@ def test_paths_between(self):
             },
         )
 
+    def test_resolve(self):
+        # Given
+        d = {
+            'nodes': node_dicts(4),
+            'edges': edge_dicts((0, 1), (0, 2), (1, 2), (1, 3), (2, 3), (3, 0)),
+        }
+        cfg = KCFG.from_dict(d)
+
+        self.assertEqual(node(1), cfg.node('d33...d8'))
+        self.assertEqual(node(1), cfg.node('d33..d8'))
+        self.assertEqual(node(1), cfg.node(node(1).id))
+
+        # Matches no nodes
+        with self.assertRaisesRegex(ValueError, 'Unknown node: deadbeef\\.\\.\\.d8'):
```

Comment:
Since you are checking for equality of the string representation, consider using `assertRaises` for readability.

Also, consider

```suggestion
        with self.assertRaisesRegex(ValueError, r'Unknown node: deadbeef\.\.\.d8'):
```